pkg/security/api/v1/types.go

```diff
@@ -0,0 +1,86 @@
+package v1
+
+import (
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	kapi "k8s.io/kubernetes/pkg/api/v1"
+)
+
+// PodSecurityPolicySubjectReview checks whether a particular user/SA tuple can create the PodSpec.
+type PodSecurityPolicySubjectReview struct {
+	unversioned.TypeMeta `json:",inline"`
+
+	// Spec defines specification for the PodSecurityPolicySubjectReview.
+	Spec PodSecurityPolicySubjectReviewSpec `json:"spec"`
+
+	// Status represents the current information/status for the PodSecurityPolicySubjectReview.
+	Status PodSecurityPolicySubjectReviewStatus `json:"status,omitempty"`
+}
+
+// PodSecurityPolicySubjectReviewSpec defines specification for PodSecurityPolicySubjectReview
+type PodSecurityPolicySubjectReviewSpec struct {
+	// PodSpec is the PodSpec to check.
+	PodSpec kapi.PodSpec `json:"podSpec"`
+
+	// User is the user you're testing for.
+	// If you specify "User" but not "Group", then is it interpreted as "What if User were not a member of any groups.
+	// If User and Groups are empty, then the check is performed using *only* the ServiceAccountName in the PodSpec.
+	User string `json:"user,omitempty"`
+
+	// Groups is the groups you're testing for.
+	Groups []string `json:"groups,omitempty"`
+}
+
+// PodSecurityPolicySubjectReviewStatus contains information/status for PodSecurityPolicySubjectReview.
+type PodSecurityPolicySubjectReviewStatus struct {
+	// AllowedBy is a reference to the rule that allows the PodSpec.  A `nil`, indicates that it was denied.
```

Comment:
Specify that this "rule" is a security context constraint or pod security policy.